tests/support/client.rs

```diff
@@ -95,12 +111,12 @@ impl Client {
     pub fn request_async(
         &self,
         builder: &mut http::request::Builder,
-    ) -> Box<Future<Item = Response, Error = String> + Send> {
+    ) -> Box<Future<Item = Response, Error = ClientError> + Send> {
         self.send_req(builder.body(Bytes::new()).unwrap())
     }
 
-    pub fn request(&self, builder: &mut http::request::Builder) -> Response {
-        self.request_async(builder).wait().expect("response")
+    pub fn request(&self, builder: &mut http::request::Builder) -> Result<Response, ClientError> {
+        self.request_async(builder).wait()
```

Comment:
I'm not sure if I understand why the `expect` was removed here. Removing it requires changing a bunch of other code (e.g., a lot of the `transparency` tests) that otherwise would not need to be removed. 

If the intention is to assert that some requests will fail (such as when the proxy cannot yet accept TLS connections), can't we just use 
```rust
client.request_async(...).wait()
```
 in those specific cases?